packages/dev/core/src/Materials/greasedLinePluginMaterial.ts

```diff
@@ -569,80 +646,191 @@ export class GreasedLinePluginMaterial extends MaterialPluginBase {
      */
-    public setSizeAttenuation(value: boolean) {
-        this._options.sizeAttenuation = value;
+    set sizeAttenuation(value: boolean) {
+        this._sizeAttenuation = value;
         this.markAllDefinesAsDirty();
     }
 
+    /**
+     * Gets the color of the line
+     */
+    get color() {
+        return this.color;
+    }
+
+    /**
+     * Sets the color of the line
+     * @param value Color3 or undefined to clear the color. You need to clear the color if you use colors and useColors = true
+     */
+    set color(value: Color3 | undefined) {
+        this.setColor(value);
+    }
+
     /**
      * Sets the color of the line. If set the whole line will be mixed with this color according to the colorMode option.
      * @param value color
      */
     public setColor(value: Color3 | undefined, doNotMarkDirty = false) {
-        if ((this._options.color === undefined && value !== undefined) || (this._options.color !== undefined && value === undefined)) {
-            this._options.color = value;
+        if ((this._color === undefined && value !== undefined) || (this._color !== undefined && value === undefined)) {
+            this._color = value;
             !doNotMarkDirty && this.markAllDefinesAsDirty();
         } else {
-            this._options.color = value;
+            this._color = value;
         }
     }
 
+    /**
+     * Get the mixing mode of the color paramater.
+     * @see GreasedLineMeshColorMode
+     */
+    get colorMode() {
+        return this._colorMode;
+    }
+
     /**
      * Sets the mixing mode of the color paramater. Default value is GreasedLineMeshColorMode.SET
      * @see GreasedLineMeshColorMode
      * @param value color mode
      */
-    public setColorMode(value: GreasedLineMeshColorMode) {
-        this._options.colorMode = value;
+    set colorMode(value: GreasedLineMeshColorMode) {
+        this._colorMode = value;
+    }
+
+    /**
+     * Gets the color distributiopn type
+     */
+    get colorsDistributionType() {
+        return this._colorsDistributionType;
+    }
+
+    /**
+     * Sets the color distribution type
+     * @see GreasedLineMeshColorDistributionType
+     * @param value color distribution type
+     */
+    set colorsDistributionType(value: GreasedLineMeshColorDistributionType) {
+        this._colorsDistributionType = value;
+        this.markAllDefinesAsDirty();
+    }
+
+    /**
+     * Gets the type of sampling of the colors texture.
+     */
+    get colorsSampling() {
+        return this._colorsSampling;
+    }
+
+    /**
+     * Sets how to sample the colors texture. The values are the same when use with textures.
+     * @param value sampling type
+     */
+    set colorsSampling(value: number) {
+        this._colorsSampling = value;
+    }
+
+    /**
+     * Gets the resolution
+     */
+    get resolution() {
+        return this._resolution;

```

Comment:
same here make resolution public

Reply:
I've introduced a private property called `aspect` which is calculated from the public `resolution` property so it has a side effect already so I didn't remove the setter/getter here.